cmd/protoc-gen-go-hessian2/generate/generate.go

```diff
@@ -0,0 +1,156 @@
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package generate
+
+import (
+	"fmt"
+)
+
+import (
+	"google.golang.org/protobuf/compiler/protogen"
+	"google.golang.org/protobuf/reflect/protoreflect"
+	"google.golang.org/protobuf/types/descriptorpb"
+)
+
+func GenHessian2(gen *protogen.Plugin, file *protogen.File) {
+	filename := file.GeneratedFilenamePrefix + ".hessian2.go"
+	g := gen.NewGeneratedFile(filename, file.GoImportPath)
+
+	g.P("package ", file.GoPackageName)
+	g.P()
+
+	g.P("import hessian \"github.com/apache/dubbo-go-hessian2\"")
+	g.P()
+
+	for _, message := range file.Messages {
+		genMessage(g, message)
+	}
+	genRegisterInitFunc(g, file)
+}
+
+func genMessage(g *protogen.GeneratedFile, m *protogen.Message) {
+	if m.Desc.IsMapEntry() {
+		return
+	}
+	g.AnnotateSymbol(m.GoIdent.String(), protogen.Annotation{
+		Location: m.Location,
+		Semantic: descriptorpb.GeneratedCodeInfo_Annotation_SET.Enum(),
+	})
+
+	g.P("type ", m.GoIdent, " struct {")
+
+	genMessageFields(g, m)
+	g.P("}")
+	g.P()
+
+	genMessageRelatedMethods(g, m)
+}
+
+func genMessageFields(g *protogen.GeneratedFile, m *protogen.Message) {
+	for _, field := range m.Fields {
+		genMessageField(g, m, field)
+	}
+}
+
+func genMessageField(g *protogen.GeneratedFile, m *protogen.Message, field *protogen.Field) {
+	goType, pointer := fieldGoType(g, field)
+	if pointer {
+		goType = "*" + goType
+	}
+
+	name := field.GoName
+	g.AnnotateSymbol(m.GoIdent.GoName+"."+name, protogen.Annotation{
+		Location: field.Location,
+		Semantic: descriptorpb.GeneratedCodeInfo_Annotation_SET.Enum(),
+	})
+	g.P(name, " ", goType)
+}
+
+func genMessageRelatedMethods(g *protogen.GeneratedFile, m *protogen.Message) {
+	g.P("func ", "(x *", m.GoIdent.GoName, ")", "JavaClassName() string {")
+	// TODO(Yuukirn): get class name by extend field
+	g.P("	return ", "\"org.test.service.", m.GoIdent.GoName, "\"")
+	g.P("}")
+
+	g.P()
+
+	g.P("func ", "(x *", m.GoIdent.GoName, ")", "String() string {")
+	g.P("	e := hessian.NewEncoder()")
+	g.P("	err := e.Encode(x)")
+	g.P("	if err != nil {")
+	g.P("		return \"\"")
+	g.P("	}")
+	g.P("	return string(e.Buffer())")
+	g.P("}")
+	g.P()
+}
+
+func genRegisterInitFunc(g *protogen.GeneratedFile, f *protogen.File) {
+	g.P("func init() {")
+	for _, message := range f.Messages {
+		g.P("hessian.RegisterPOJO(new(", message.GoIdent.GoName, "))")
+	}
+	g.P("}")
+	g.P()
+}
+
+// fieldGoType returns the Go type used for a field.
+//
+// If it returns pointer=true, the struct field is a pointer to the type.
+func fieldGoType(g *protogen.GeneratedFile, field *protogen.Field) (goType string, pointer bool) {
+	if field.Desc.IsWeak() {
+		return "struct{}", false
+	}
+
+	pointer = field.Desc.HasPresence()
+	switch field.Desc.Kind() {
```

Comment:
We should discuss the mappings in details.
type serialization conversion: proto type => go type => java type
helping tool conversion: java interface file => proto file
e.g. Java does not support unsigned basic type, it would be complex when implementing helping tool if we decide to support unsigned basic type in proto and go.